### PR TITLE
[dashboard] handle /from-referrer

### PR DESCRIPTION
**Pull request by svenefftinge**
Fix https://github.com/gitpod-io/gitpod/issues/3682

### CODE REVIEW

**Comment by gtsiolis** [comment]
Looking at this now! 👀

**Comment by svenefftinge** [comment]
finishing this now

**Comment by svenefftinge** [comment]
Test e.g. by opening this PR, through:

https://sefftinge-dashboard-from-referrer-3682.staging.gitpod-dev.com/from-referrer

File: components/dashboard/src/FromReferrer.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        url.hash = contextUrl;
+        window.location.href = url.toString();
+        return <></>;
+    }
+    return <div className="lg:px-28 px-10 flex flex-col space-y-2">
+        <div className="px-6 py-3 flex justify-between space-x-2 text-gray-400 h-96">
+            <div className="flex flex-col items-center w-96 m-auto mt-40">
+                <h3 className="text-center pb-3 text-gray-500">No Referrer Found</h3>
+                <div className="text-center pb-6 text-gray-500">
+                    <p>It looks like you are trying to open a workspace, but the referrer URL is empty. This happens when the git hoster doesn't send the referrer header.
```
Comment (gtsiolis) [inline components/dashboard/src/FromReferrer.tsx:24]: **thought**: I still find the term _git hoster_ quite confusing. We could drop this sentence but this is fine, too.

download

**Review by gtsiolis — APPROVED** [review]
DOES. THE. JOB. Let's ship this! 🚢

**Pull request merged**